SMG Rebalance and Refactoring

Coureere opened the pull request:
## About The Pull Request
This PR was made to address issue of all SMGs we had in codebase was more or less the same and where was almost no difference between them. This is my take on solving this.



## Details
- Added new firemodes. `FULL_AUTO_300` (delay 2) replaced `FULL_AUTO_250` because it was unused and `FULL_AUTO_800` (delay 0,8) was added.
- Idaho SMG was remade into long awaited .40 Magnum SMG. Core of this gun is "fire slow, hit hard". Feeds off newly added .40 SMG mags with capacity of 25. Sprites by @shazbot194.
- Straylight was altered in a way that this is even more of a "spray-and-pray" gun that it was before. Being IH issued SMG, it's now more fit to be used with rubber rounds for swift takedowns. High firerate comes at cost of reduced damage and penetration. Still quite deadly with lethal rounds due to humongous fire rate. `FULL_AUTO_800` was made specifically for it.
- Triple of existing SMGs - Atreides, C20r and Drozd will form line of generic universal-purpose SMGs which doesn't have anything special in particular, with following one (in given order) being direct upgrade of a previous one. Atreides would be available as general civilian SMG that you could buy in FS vendor, C20r would be available either from uplink for antags (it's still very decent and deadly SMG) and best of all, Drozd, would be available on one of the Excelsior disks (all three would be added as loot in upcoming PR as well).
- Atreides in particular can now attach silencer. Sprites was kitbashed myself from existing silencer from other gun. If someone can make it better - you're welcome (shazbot, don't post cursed silence gun).
- Three round bursts was universally removed from all guns affected by this PR for the reason of them being inferior to the full auto mode and never used by anyone. Less bloat.
- Fluff descriptions of guns was expanded. I would like to receive feedback on them and/or alternative versions because my grammar is rather lacking.
- Material cost of affected guns was also altered, mainly to include steel also.
- Price tags was altered on all affected guns, mainly made SMGs cheaper.
- Stats also changed universally to reflect concept described above, criticism and/or suggestions is welcome, test would also be great.

</details>

SpadesSlick commented on code/game/objects/items/weapons/autolathe_disks.dm:
```diff
@@ -821,7 +825,7 @@
 	)
 
 /obj/item/weapon/computer_hardware/hard_drive/portable/design/guns/dallas
-	disk_name = "PAR - .25 Dallas"
```
Leave the default name. Also, starting ANYTHING with number, dot or something like this is bad for readability.

Coureere replied:
Resolved.

SpadesSlick commented on code/modules/projectiles/guns/projectile/automatic/c20r.dm:
```diff
@@ -13,21 +16,21 @@
 	reload_sound 	= 'sound/weapons/guns/interact/sfrifle_magin.ogg'
 	cocked_sound 	= 'sound/weapons/guns/interact/sfrifle_cock.ogg'
-	zoom_factor = 0.2
-	damage_multiplier = 1.1
```
Why ``damage_multiplier`` was changed to ``penetration_multiplier``? Even is it's 1.0, leave it just like that for code readability.

Coureere replied:
Adressed.

SpadesSlick commented on code/modules/projectiles/guns/projectile/automatic/idaho.dm:
```diff
@@ -1,35 +1,37 @@
 	magazine_type = /obj/item/ammo_magazine/smg
-	matter = list(MATERIAL_PLASTEEL = 16, MATERIAL_WOOD = 4)
-	price_tag = 1600
-	damage_multiplier = 0.9
```
Same as above.

Coureere replied:
Adressed.

SpadesSlick commented on code/modules/projectiles/guns/projectile/automatic/straylight.dm:
```diff
@@ -13,15 +17,18 @@
 	auto_eject = 1
-	matter = list(MATERIAL_PLASTEEL = 16, MATERIAL_PLASTIC = 4)
-	price_tag = 2500 //good smg with normal recoil and silencer possibility
+	matter = list(MATERIAL_PLASTEEL = 7, MATERIAL_STEEL = 3, MATERIAL_PLASTIC = 10)
```
I understand that 16 is a lot, but 7 is too small number.

Coureere replied:
Adressed, changed to 12, 2, 8.

SpadesSlick commented on code/modules/projectiles/guns/projectile/automatic/drozd.dm:
```diff
@@ -1,22 +1,24 @@
-	price_tag = 2000
-	damage_multiplier = 0.75
-	recoil_buildup = 1.5
+	matter = list(MATERIAL_PLASTEEL = 12, MATERIAL_STEEL = 4, MATERIAL_PLASTIC = 4)
```
Individually those numbers make sense, together this gun is a meme mess.

Coureere replied:
My following proposal is to make it more or less identical to C20r so it would be just excelsior skin of general purpose overall good SMG. If you want to nerf something in particular about it - point it out.

Coureere replied:
Should be resolved now. Made it rare AP-focused SMG that you can only find on disk from advanced disk spawners for now. Later would add as very rare gun loot itself.

MadmanMartian commented on code/modules/projectiles/guns/projectile/automatic/atreides.dm:
```diff
@@ -1,6 +1,7 @@
 	name = "FS SMG .35 Auto \"Atreides\""
-	desc = "The Atreides is a replica of an old and popular SMG. It has a strong kick. Uses .35 Auto rounds."
+	desc = "The Atreides is a replica of an old and popular SMG. Cheap and mass produced generic self-defence weapon. \
+			Design is so generic that all of it characteristics is neither too bad or too good. Uses .35 Auto rounds."
```
"The overall design is so generic that it is neither considered good nor bad in comparison to other firearms. Uses .35 Auto rounds."

Coureere replied:
Many thanks, implemented.

MadmanMartian commented on code/modules/projectiles/guns/projectile/automatic/drozd.dm:
```diff
@@ -1,22 +1,26 @@
 /obj/item/weapon/gun/projectile/automatic/drozd
 	name = "Excelsior .35 Auto \"Drozd\""
-	desc = "An excellent fully automatic submachinegun. Famous for it's perfomance in close quarters. Uses .35 Auto rounds."
+	desc = "An excellent fully automatic submachinegun. Made by \"Excelsior\" by reverse-engineering and improving upon Moebius \
```
"An excellent fully automatic submachine gun, Made by \"Excelsior\", reverse engineered from the \"Moebius"\ C20r, with an emphasis on armor penetration over direct damage. Uses .35 Auto rounds."

Coureere replied:
Thing is Moebius is also reverse-engineered C20r, so it's reverse-engineered from reverse-engineered kind of joke. We already have similar joke in AK description.

Coureere replied:
Replaced my mess with this.

MadmanMartian commented on code/modules/projectiles/guns/projectile/automatic/idaho.dm:
```diff
@@ -1,35 +1,38 @@
-	name = "FS SMG .35 Auto \"Idaho\""
-	desc = "The Idaho is a cheap self-defence weapon, mass-produced by \"Frozen Star\" for paramilitary and private use. Uses 30 Auto rounds."
+	name = "FS SMG .40 Magnum \"Idaho\""
+	desc = "The Idaho is an experimental SMG made by \"Frozen Star\" for paramilitary and private use. \
```
"An experimental Submachine gun made by \"Frozen Star\", for Paramilitary and private security use. Rifled to take a larger caliber than a typical submachine gun, it boasts a greater impact, but suffers from poor recoil control as a result. Uses .40 Magnum rounds."

Coureere replied:
Good one.

MadmanMartian commented on code/modules/projectiles/guns/projectile/automatic/molly.dm:
```diff
@@ -1,6 +1,10 @@
 /obj/item/weapon/gun/projectile/automatic/molly
 	name = "FS MP .35 Auto \"Molly\""
-	desc = "An experimental fully automatic pistol. Compact and flexible, but somewhat underpowered. Issued to non-combatants among Ironhammer as powerful self-protection sidearm. Custom magwell allows it to feed both from highcap pistol and SMG magazines. Uses .35 Auto rounds."
+	desc = "An experimental fully automatic pistol. Was designed as a middle ground between SMGs and pistols to put superior \
```
"An experimental fully automatic pistol, designed as a middle ground between SMGs and Pistols. Primarily employed in CQC scenarios or as a civilian self defence tool. Takes both .35 Highcap and .35 Auto rounds."

Coureere replied:
Used it, thanks.

MadmanMartian commented on code/modules/projectiles/guns/projectile/automatic/straylight.dm:
```diff
@@ -1,6 +1,10 @@
 /obj/item/weapon/gun/projectile/automatic/straylight
 	name = "FS SMG .35 Auto \"Straylight\""
-	desc = "A compact and lightweight submachinegun that sprays small rounds rapidly. Sacrifices a fire selector to cut mass, so it requires a careful hand. Uses .35 Auto rounds."
+	desc = "A compact and lightweight submachinegun that sprays small rounds rapidly. It supposed to be testing ground for newly \
```
"A compact, lightweight rapid submachine gun. Primarily used for testing ammunition and weapon modifications. Suffers from poor recoil control, and underperforming ballistic impact, but makes up for this through sheer firerate. Uses .35 Auto rounds."

Coureere replied:
Encorporated.

Coureere commented:
Drozd is now .40 Magnum heavy smg available only from lootable gun disk and is AP-focused gun. Idaho was renamed to Zoric and was given to serbs, as well as it remains in loot and lootable as gun disk.

Coureere commented:
Means of Production (and Means of Revolution too) disk stuff if reordered to make much more sense. Added missing ammo (packets mostly), added rubber ammo for each gun also.
